plugin/storage/es/mappings/mapping.go

```diff
@@ -9,36 +9,58 @@ import (
 	"strings"
 
 	"github.com/jaegertracing/jaeger/pkg/es"
+	"github.com/jaegertracing/jaeger/pkg/es/config"
 )
 
 // MAPPINGS contains embedded index templates.
 //
 //go:embed *.json
 var MAPPINGS embed.FS
 
-// MappingBuilder holds parameters required to render an elasticsearch index template
+// MappingBuilder holds common parameters required to render an elasticsearch index template
 type MappingBuilder struct {
-	TemplateBuilder              es.TemplateBuilder
-	Shards                       int64
-	Replicas                     int64
-	PrioritySpanTemplate         int64
-	PriorityServiceTemplate      int64
-	PriorityDependenciesTemplate int64
-	PrioritySamplingTemplate     int64
-	EsVersion                    uint
-	IndexPrefix                  string
-	UseILM                       bool
-	ILMPolicyName                string
+	TemplateBuilder es.TemplateBuilder
+	Indices         config.Indices
+	EsVersion       uint
+	UseILM          bool
+	ILMPolicyName   string
+}
+
+// IndexTemplateOptions holds parameters required to render an elasticsearch index template
+type IndexTemplateOptions struct {
+	UseILM        bool
+	ILMPolicyName string
+	config.IndexOptions
```

Comment:
nit: I would recommend against embedding the whole struct, because it makes the templates tightly coupled with the configuration. E.g. it looks like only a couple of fields are actually used in the templates, prefix and priority, so I would define them explicitly in IndexTemplateOptions and copy from IndexOptions as needed. This way we won't have to change the templates just because we changed the config struct.